src/coprocessor/dag/expr/builtin_op.rs

```diff
@@ -0,0 +1,376 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// TODO: remove following later
+#![allow(dead_code)]
+
+use super::{FnCall, Result, StatementContext};
+use coprocessor::codec::Datum;
+
+impl FnCall {
+    pub fn logic_and(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        let arg0 = try!(self.children[0].eval_int(ctx, row));
+        let arg1 = try!(self.children[1].eval_int(ctx, row));
+        match (arg0, arg1) {
```

Comment:
If `arg0` is `Some(0)` or `None`, you can return early without evaluating `arg1`. And I prefer the name `lhs` and `rhs`.